packages/medusa/src/services/payment-provider.ts

```diff
@@ -0,0 +1,597 @@
+import { Payment } from "@models/payment"
+import { FindConfig, Selector } from "../types/common"
+import { PaymentSession, PaymentSessionStatus } from "@models/payment-session"
+import { Cart } from "@models/cart"
+import { AwilixContainer } from "awilix"
+import { Refund } from "@models/refund"
+
+type InjectedDependencies = {
+  manager: EntityManager
+  paymentSessionRepository: typeof PaymentSessionRepository
+  paymentProviderRepository: typeof PaymentProviderRepository
+  paymentRepository: typeof PaymentRepository
+  refundRepository: typeof RefundRepository
+}
+
+/**
+ * Helps retrieve payment providers
+ */
+export default class PaymentProviderService extends TransactionBaseService<
+  PaymentProviderService,
+  InjectedDependencies
+> {
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+  protected readonly container_: AwilixContainer<InjectedDependencies>["cradle"]
+  protected readonly paymentSessionRepository_: typeof PaymentSessionRepository
+  protected readonly paymentProviderRepository_: typeof PaymentProviderRepository
+  protected readonly paymentRepository_: typeof PaymentRepository
+  protected readonly refundRepository_: typeof RefundRepository
+
+  constructor(container: AwilixContainer<InjectedDependencies>["cradle"]) {
+    super(container)
+
+    this.container_ = container
+    this.manager_ = container.manager
+    this.paymentSessionRepository_ = container.paymentSessionRepository
+    this.paymentProviderRepository_ = container.paymentProviderRepository
+    this.paymentRepository_ = container.paymentRepository
+    this.refundRepository_ = container.refundRepository
+  }
+
+  async registerInstalledProviders(providerIds): Promise<void> {
+    return await this.atomicPhase_(
+      async (transactionManager: EntityManager) => {
+        const model = transactionManager.getCustomRepository(
+          this.paymentProviderRepository_
+        )
+        await model.update({}, { is_installed: false })
+
+        await Promise.all(
+          providerIds.map((providerId) => {
+            const provider = model.create({
+              id: providerId,
+              is_installed: true,
+            })
+            return model.save(provider)
+          })
+        )
+      }
+    )
+  }
+
+  async list(): Promise<PaymentProvider[]> {
+    return await this.atomicPhase_(
+      async (transactionManager: EntityManager) => {
+        const ppRepo = transactionManager.getCustomRepository(
+          this.paymentProviderRepository_
+        )
+        return await ppRepo.find()
+      }
+    )
+  }
+
+  async retrievePayment(
+    id: string,
+    relations: string[] = []
+  ): Promise<Payment> {
+    return await this.atomicPhase_(
+      async (transactionManager: EntityManager) => {
+        const paymentRepo = transactionManager.getCustomRepository(
+          this.paymentRepository_
+        )
+        const validatedId = validateId(id)
```

Comment:
todo(non-blocking): Remove since config is not passed i.e. simply returning the provided id.